sys/vm/vm_map.c

```diff
@@ -201,7 +201,15 @@ vm_map_log(const char *prefix, vm_map_entry_t entry)
 		prt[2] = 'x';
 	else
 		prt[2] = '-';
-	prt[3] = '\0';
+	if (entry->protection & VM_PROT_READ_CAP)
+		prt[0] = 'R';
+	else
+		prt[0] = '-';
+	if (entry->protection & VM_PROT_WRITE_CAP)
+		prt[1] = 'W';
+	else
+		prt[1] = '-';
+	prt[5] = '\0';
```

Comment:
So in general I haven't put the bits or things like this under the #ifdef as it would add a lot of #ifdef's.  The permissions are harmless for non-CHERI platforms.

Reply:
I think it's also true that to the extent that we can make things similar for x86 developers that's for the best.